server/mediapublic/mediapublic/models.py

```diff
@@ -259,8 +261,59 @@ def to_dict(self):
             secondary_website = self.secondary_website,
             creation_datetime = str(self.creation_datetime),
         )
-        return resp    
-    
+        return resp
+
+class PlaylistAssignments(Base, CreationMixin):
+    __tablename__ = 'playlist_assignments'
+    id = Column(Integer, primary_key=True)
+    playlist_id = Column(Integer, ForeignKey('playlists.id'))
+    recording_id = Column(Integer, ForeignKey('recordings.id'))
+
+    @classmethod
+    def get_by_playlist_id_and_recording_id(cls, pid, rid):
+        with transaction.manager:
+            playlists = DBSession.query(
+                PlaylistAssignments,
+            ).filter(
+                PlaylistAssignments.playlist_id == pid,
+                PlaylistAssignment.recording_id == rid,
+            ).all()
+        return assignments[0] # combination should be unique
+
+    def to_dict(self):
+        resp = dict(
+            id = self.id,
+            playlist_id = self.playlist_id,
+            recording_id = self.recording_id,
+        )
+        return resp
+
+class Playlists(Base, CreationMixin):
+
+    __tablename__ = 'playlists'
+    id = Column(Integer, primary_key=True)
+    author_id = ReqColumn(Integer, ForeignKey('people.id'))
+    title = ReqColumn(UnicodeText)
+    recordings = relationship("Recordings", secondary=PlaylistAssignments.__table__, backref="playlists")
+
+    @classmethod
+    def get_by_user_id(cls, id):
+        with transaction.manager:
+            playlists = DBSession.query(
+                Playlists,
+            ).filter(
+                Playlists.author_id == id,
+            ).all()
+        return playlists
+
+    def to_dict(self):
```

Comment:
We may want to include `self.recordings` here as `recordings = [r.to_dict() for r in self.recordings]`, but let's see what the actual need of the UI is first.